### PR TITLE
docs: add release notes for 20.1.0

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## Issue being fixed or feature implemented
Added release notes for 20.1.0

## What was done?
Wrote release notes from WIP release-notes; trimmed some fat from release notes

## How Has This Been Tested?
Ran linters

## Breaking Changes
None; docs only

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
+
+Modified the Dockerfile to exclude the `dash-qt` graphical wallet interface from the set of binaries copied to
+`/usr/local/bin` during the Docker image build process. This change streamlines the Docker image, making it more
+suitable for server environments and headless applications, where the graphical user interface of `dash-qt` is not
+required. The update enhances the Docker image's efficiency by reducing its size and minimizing unnecessary components.
 
-The miner has been disabled for Windows binaries built via Guix. This resolves
-an issue that was preventing users from installing v20.0.3 due to false
-positive alerts from Microsoft Defender and other antivirus providers.
+## Command-line options
```
Comment (UdjinM6) [inline doc/release-notes.md:83]: 2 new options:
`-capturemessages` Capture all P2P messages to disk, see contrib/message-capture/message-capture-docs.md for more info
`-startupnotify=<cmd>` Execute command on startup.

File: doc/release-notes.md [hunk 2]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
+Deterministic (HD) wallets by default. This change aligns the behavior of `dashd` and `dash-qt` with the previously
+optional `-usehd=1` flag, making HD wallets the standard for all users.
+
+While HD wallets are now enabled by default to improve user experience, Dash Core still allows for the creation of
+non-HD wallets by using the `-usehd=0` flag. However, users should be aware that this option is intended for legacy
+support and will be removed in future releases. Importantly, even with an HD wallet, users can still import non-HD
+private keys, ensuring flexibility in managing their funds.
+
+## Configuration option changes
+
```
Comment (UdjinM6) [inline doc/release-notes.md:43]: that's what `Command-line options` section describes, right?
```suggestion
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by knst — CHANGES_REQUESTED** [review]

File: doc/release-notes.md [hunk 3]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
+the peer. It will return one of six options. For more information, see the `getpeerinfo` help documentation.
+- The `getpeerinfo` RPC now has additional `last_block` and `last_transaction` fields that return the UNIX epoch time
+of the last block and the last valid transaction received from each peer. (#19731)
+- `getblockchaininfo` now returns a new `time` field, that provides the chain tip time.
+- `fundrawtransaction` and `walletcreatefundedpsbt` when used with the `lockUnspents` argument now lock manually
+selected coins, in addition to automatically selected coins. Note that locked coins are never used in automatic coin
+selection, but can still be manually selected.
+- The `testmempoolaccept` RPC returns `vsize` and a `fee` object with the `base` fee if the transaction passes
+validation. The `testmempoolaccept` RPC now accepts multiple transactions (still experimental at the moment, API may be
+unstable).
```
Comment (thephez) [inline doc/release-notes.md:64]: Maybe make these all consistent?
```suggestion
- `getpeerinfo`: now returns a `connection_type` field. This indicates the type of connection established with
the peer. It will return one of six options. For more information, see the `getpeerinfo` help documentation.
- `getpeerinfo`: now has additional `last_block` and `last_transaction` fields that return the UNIX epoch time
of the last block and the last valid transaction received from each peer. (#19731)
- `getblockchaininfo`: now returns a new `time` field, that provides the chain tip time.
- `fundrawtransaction` and `walletcreatefundedpsbt`: when used with the `lockUnspents` argument now lock manually
selected coins, in addition to automatically selected coins. Note that locked coins are never used in automatic coin
selection, but can still be manually selected.
- `testmempoolaccept`: returns `vsize` and a `fee` object with the `base` fee if the transaction passes
validation. The `testmempoolaccept` RPC now accepts multiple transactions (still experimental at the moment, API may be
unstable).
```

File: doc/release-notes.md [hunk 4]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
+
+## Added RPCs
+
+- `submitchainlock` RPC allows the submission of a ChainLock signature. Note: This RPC is whitelisted for the Platform
+RPC user.
+- `getassetunlockstatuses` RPC allows fetching of Asset Unlock txs by their withdrawal index. The RPC accepts an array
+of indexes and an optional block height. The possible outcomes per each index are: `chainlocked`, `mined`, `mempooled`,
+`unknown`. Note: If a specific block height is passed on request, then only `chainlocked` and `unknown` outcomes are
+possible. This RPC is whited for the Platform RPC user.
+- `quorum dkginfo` RPC returns information about DKGs: `active_dkgs` and `next_dkg`.
```
Comment (thephez) [inline doc/release-notes.md:74]:
```suggestion
- `submitchainlock`: allows the submission of a ChainLock signature. Note: This RPC is whitelisted for the Platform
RPC user.
- `getassetunlockstatuses`: allows fetching of Asset Unlock txs by their withdrawal index. The RPC accepts an array
of indexes and an optional block height. The possible outcomes per each index are: `chainlocked`, `mined`, `mempooled`,
`unknown`. Note: If a specific block height is passed on request, then only `chainlocked` and `unknown` outcomes are
possible. This RPC is whitelisted for the Platform RPC user.
- `quorum dkginfo`: returns information about DKGs: `active_dkgs` and `next_dkg`.
```

File: doc/release-notes.md [hunk 5]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
 Thanks to everyone who directly contributed to this release:
 
-- Konstantin Akimov (knst)
+- Kittywhiskers Van Gogh
+- Konstantin Akimov
+- Odysseas Gabrielides
 - PastaPastaPasta
+- thephez
 - UdjinM6
+- Vijay Manikpuri
```
[the remark this answers is not in the document]
Comment (UdjinM6) [reply doc/release-notes.md:124]: Good catch! https://github.com/dashpay/dash/pulls?q=is%3Apr+author%3Apanleone+is%3Aclosed+milestone%3A20.1 https://github.com/panleone Alessandro Rezzi

File: doc/release-notes.md [hunk 6]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
+## RPC changes
+
+- `protx info`: It's now possible to get historical data at a specific block by providing its blockhash.
+- RPC `gettxchainlocks` will also return the status `mempool` indicating whether the transaction is in the mempool 
+or not.
+- Exposed transaction version numbers are now treated as unsigned 16-bit integers instead of signed 16-bit integers.
+This matches their treatment in consensus logic. Versions greater than 3 continue to be non-standard (matching previous
+behavior of smaller than 1 or greater than 3 being non-standard). Note that this includes the joinpsbt command, which
+combines partially-signed transactions by selecting the highest version number.
+- New RPC `getrawtransactionmulti` that can return batch up to 100 raw transactions at one request.
```
Comment (thephez) [inline doc/release-notes.md:53]: Also, this RPC should be moved down to the "Added RPCs" section
```suggestion
- New RPC `getrawtransactionmulti` that can return a batch of up to 100 raw transactions per request.
```

File: doc/release-notes.md [hunk 7]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
 
-Updated expired gpg keys for builders. Note that the actual keys themselves are the same.
-The updated keys changed their expiration dates and added additional information such as
-new email addresses.
+## Notification changes
 
-## Governance
+- `-walletnotify` notifications are now sent for wallet transactions that are removed from the mempool because they
+conflict with a new block. These notifications were sent previously before the v18.1 release, but had been broken since
+that release.
```
Comment (thephez) [inline doc/release-notes.md:92]:
```suggestion
- `-walletnotify` notifications are now sent for wallet transactions that are removed from the mempool because they
conflict with a new block. These notifications were sent previously, but have been broken since
the v18.1 release.
```

**Review by thephez — COMMENTED** [review]
Some suggestions

File: doc/release-notes.md [hunk 8]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
 
-## Asset lock and unlock transactions
+The `--enable-upnp-default` and `--enable-natpmp-default` options have been removed. If you want to use port mapping,
+you can configure it using a `.conf` file, or by passing the relevant options at runtime.
 
-Use the correct quorum for signing withdrawal (asset unlock) txes on RegTest.
+## Wallet
 
-# v20.0.4 Change log
+- The wallet can create a transaction without change even when the keypool is empty. Previously it failed.
```
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply doc/release-notes.md:103]: Why didn't they have WIP release notes!?!

File: doc/release-notes.md [hunk 9]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
+## Added RPCs
+
+- `submitchainlock`: allows the submission of a ChainLock signature. Note: This RPC is whitelisted for the Platform
+RPC user.
+- `getassetunlockstatuses`: allows fetching of Asset Unlock txs by their withdrawal index. The RPC accepts an array
+of indexes and an optional block height. The possible outcomes per each index are: `chainlocked`, `mined`, `mempooled`,
+`unknown`. Note: If a specific block height is passed on request, then only `chainlocked` and `unknown` outcomes are
+possible. This RPC is whitelisted for the Platform RPC user.
+- `quorum dkginfo`: returns information about DKGs: `active_dkgs` and `next_dkg`.
+- `getrawtransactionmulti`: that can return a batch of up to 100 raw transactions per request.
```
Comment (thephez) [inline doc/release-notes.md:74]:
```suggestion
- `getrawtransactionmulti`: allows requesting a batch of up to 100 raw transactions per request.
```

File: doc/release-notes.md [hunk 10]
```diff
@@ -35,65 +28,109 @@ reindex or re-sync the whole chain.
...
+In this release, we are taking a significant step towards enhancing the Dash wallet's usability by enabling Hierarchical
+Deterministic (HD) wallets by default. This change aligns the behavior of `dashd` and `dash-qt` with the previously
+optional `-usehd=1` flag, making HD wallets the standard for all users.
+
+While HD wallets are now enabled by default to improve user experience, Dash Core still allows for the creation of
+non-HD wallets by using the `-usehd=0` flag. However, users should be aware that this option is intended for legacy
+support and will be removed in future releases. Importantly, even with an HD wallet, users can still import non-HD
+private keys, ensuring flexibility in managing their funds.
+
+## Configuration option changes
```
Comment (thephez) [inline doc/release-notes.md:42]: Suggestion on @knst's suggestion :slightly_smiling_face: 
```suggestion
## Windows release signing

Windows releases are signed by a new certificate that replaces the expired one used previously.
```

File: doc/release-notes.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (knst) [reply doc/release-notes.md:103]: @thephez try on testnet:
Copy in clipboard (or save in file):
```
cHNidP8BADMCAAAAAdSfUVEPM7KznCV+HnT+rDxtfu42z/QAeAIz+HfGfO9CAAAAAAD/////AAAAAAAAAA==
```
Click "File -> Load from clipboard"

[the remark this answers is not in the document]
Comment (knst) [reply doc/release-notes.md:103]: @thephez you still can't create PSBT transaction with GUI only, but you can sign it with GUI now.
I used this CLI command to create PSBT transaction:
```
createpsbt '[{"txid":"42ef7cc677f833027800f4cf36ee7e6d3cacfe741e7e259cb3b2330f51519fd4", "vout":0}]' '[]'
```

[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply doc/release-notes.md:103]: Hmm; feels like we shouldn't add this info here then. A bit too technical still IMO

[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply doc/release-notes.md:103]: > there's several changes from backports that didn't have dedicated release note in bitcoin but maybe useful to mention that

@thephez I updated the suggestion here: please review

[the remark this answers is not in the document]
Comment (thephez) [reply doc/release-notes.md:103]: Change the last one to smth like:
Added a "Close all wallet" menu item for Dash-Qt

And make them all indented at the same level.

File: doc/release-notes.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (PastaPastaPasta) [reply doc/release-notes.md:132]: Done

File: doc/release-notes.md (hunk 1 above)
Comment (PastaPastaPasta) [reply doc/release-notes.md:85]: https://github.com/dashpay/dash/pull/5910/files#diff-e04cb1e9f29f76897c6b84334238f621ff45ba13cf180ad1d0d442d50ebe51f7R89-R90

File: doc/release-notes.md [hunk 11]
```diff
@@ -35,65 +28,117 @@ reindex or re-sync the whole chain.
...
-an issue that was preventing users from installing v20.0.3 due to false
-positive alerts from Microsoft Defender and other antivirus providers.
+## Command-line options
 
-Use a self-signed Windows code signing certificate instead of expired one. The
-expired certificate being used previously caused most antivirus software to trigger.
-The new self-signed certificate results in a greatly improved installation experience.
+- The same ZeroMQ notification (e.g. `-zmqpubhashtx=address`) can now be specified multiple times to publish the same
+notification to different ZeroMQ sockets.
+- `-capturemessages`: Capture all P2P messages to disk, see contrib/message-capture/message-capture-docs.md for more info
```
Comment (thephez) [inline doc/release-notes.md:89]: Is that supposed to be a link to the markdown file?

File: doc/release-notes.md [hunk 12]
```diff
@@ -35,65 +28,117 @@ reindex or re-sync the whole chain.
...
+private keys, ensuring flexibility in managing their funds.
+
+## Windows release signing
+
+Windows releases are signed by a new certificate that replaces the expired one used previously.
+
+## RPC changes
+
+- `protx info`: It's now possible to get historical data at a specific block by providing its blockhash.
+- RPC `gettxchainlocks` will also return the status `mempool` indicating whether the transaction is in the mempool
```
Comment (thephez) [inline doc/release-notes.md:49]:
```suggestion
- `gettxchainlocks`: will now return the status `mempool` indicating whether the transaction is in the mempool.
```

File: doc/release-notes.md [hunk 13]
```diff
@@ -35,65 +28,117 @@ reindex or re-sync the whole chain.
...
+or not.
+- Exposed transaction version numbers are now treated as unsigned 16-bit integers instead of signed 16-bit integers.
+This matches their treatment in consensus logic. Versions greater than 3 continue to be non-standard (matching previous
+behavior of smaller than 1 or greater than 3 being non-standard). Note that this includes the joinpsbt command, which
+combines partially-signed transactions by selecting the highest version number.
+- `getpeerinfo`: now returns a `connection_type` field. This indicates the type of connection established with
+the peer. It will return one of six options. For more information, see the `getpeerinfo` help documentation.
+- `getpeerinfo`: now has additional `last_block` and `last_transaction` fields that return the UNIX epoch time
+of the last block and the last valid transaction received from each peer.
+- `getblockchaininfo`: now returns a new `time` field, that provides the chain tip time.
```
Comment (thephez) [inline doc/release-notes.md:59]:
```suggestion
- `getblockchaininfo`: now returns a new `time` field that provides the chain tip time.
```

File: doc/release-notes.md (hunk 11 above)
Comment (PastaPastaPasta) [reply doc/release-notes.md:89]: Maybe this?
```suggestion
- `-capturemessages`: Capture all P2P messages to disk, see [contrib/message-capture/message-capture-docs.md](https://github.com/dashpay/dash/blob/v20.1.0/contrib/message-capture/message-capture-docs.md) for more info
```

Comment (knst) [reply doc/release-notes.md:89]: path can be relative so far as I know, such as `../contrib/message-capture/message-capture-docs.md`

+ `-capturemessages`: Capture all P2P messages to disk, see [message-capture-docs.md](contrib/message-capture/message-capture-docs.md) for more info

Comment (thephez) [reply doc/release-notes.md:89]: That should work (once the tag exists) :+1:

**Review by knst — APPROVED** [review]
LGTM

**Review by kwvg — APPROVED** [review]
LGTM

**Comment by UdjinM6** [comment]
I think we should also mention complete removal of legacy IS support (no `-llmqinstantsend` and `-llmqtestinstantsend` cmd-line options, no `islock` msg, no inv `30`) https://github.com/dashpay/dash/pull/5553 and protocol bump to 70231 https://github.com/dashpay/dash/pull/5905

File: doc/release-notes.md [hunk 14]
```diff
@@ -35,65 +28,117 @@ reindex or re-sync the whole chain.
...
-an issue that was preventing users from installing v20.0.3 due to false
-positive alerts from Microsoft Defender and other antivirus providers.
+## Command-line options
 
-Use a self-signed Windows code signing certificate instead of expired one. The
-expired certificate being used previously caused most antivirus software to trigger.
-The new self-signed certificate results in a greatly improved installation experience.
+- The same ZeroMQ notification (e.g. `-zmqpubhashtx=address`) can now be specified multiple times to publish the same
+notification to different ZeroMQ sockets.
+- `-capturemessages`: Capture all P2P messages to disk, see [contrib/message-capture/message-capture-docs.md](https://github.com/dashpay/dash/blob/v20.1.0/contrib/message-capture/message-capture-docs.md) for more info
```
Comment (thephez) [inline doc/release-notes.md:89]: The script to parse the captured stuff isn't working due to not understanding Dash-specific messages. Not sure we want to include this line if that can't run without crashing. It severely limits the usefulness of the captured info.

Comment (PastaPastaPasta) [reply doc/release-notes.md:89]:
```suggestion
```

**Comment by PastaPastaPasta** [comment]
> I think we should also mention complete removal of legacy IS support (no `-llmqinstantsend` and `-llmqtestinstantsend` cmd-line options, no `islock` msg, no inv `30`) #5553 and protocol bump to 70231 #5905

Would you like to propose some language around this?

**Comment by UdjinM6** [comment]
> > I think we should also mention complete removal of legacy IS support (no `-llmqinstantsend` and `-llmqtestinstantsend` cmd-line options, no `islock` msg, no inv `30`) #5553 and protocol bump to 70231 #5905
> 
> Would you like to propose some language around this?

I think smth like:
> Legacy InstantSend was deprecated with the activation of Deterministic InstantSend but we kept the underlying logic for some time to ensure smooth transition. v20.1 finally removes all of Legacy InstantSend logic and will no longer support sending and receiving `islock` p2p message and `inv` p2p message with `id` equal `30`. Such messages will be seen as `unknown` by v20.1 nodes. To make it clear that there was a change in network protocol the version was bumped to 70231. Please note that `-llmqinstantsend` and `-llmqtestinstantsend` command line options are no longer available either.

I'd like to also ask @thephez to rephrase it to sound a bit better though :)

**Comment by thephez** [comment]
> Legacy InstantSend was deprecated with the activation of Deterministic InstantSend but we kept the underlying logic for some time to ensure smooth transition. v20.1 finally removes all of Legacy InstantSend logic and will no longer support sending and receiving `islock` p2p message and `inv` p2p message with `id` equal `30`. Such messages will be seen as `unknown` by v20.1 nodes. To make it clear that there was a change in network protocol the version was bumped to 70231. Please note that `-llmqinstantsend` and `-llmqtestinstantsend` command line options are no longer available either.

Looks good. I just made a couple small changes...

Legacy InstantSend was deprecated with the activation of Deterministic InstantSend but we kept the underlying logic for some time to ensure a smooth transition. v20.1 finally removes all Legacy InstantSend logic and will no longer support sending and receiving islock p2p message and inv p2p message with id equal 30. Such messages will be seen as unknown by v20.1 nodes. The protocol version was incremented to 70231 to indicate a change in network protocol. Please note that the `-llmqinstantsend` and `-llmqtestinstantsend` command line options have also been removed as part of this change.

File: doc/release-notes.md [hunk 15]
```diff
@@ -35,65 +28,116 @@ reindex or re-sync the whole chain.
...
+
+While HD wallets are now enabled by default to improve user experience, Dash Core still allows for the creation of
+non-HD wallets by using the `-usehd=0` flag. However, users should be aware that this option is intended for legacy
+support and will be removed in future releases. Importantly, even with an HD wallet, users can still import non-HD
+private keys, ensuring flexibility in managing their funds.
+
+## Windows release signing
+
+Windows releases are signed by a new certificate that replaces the expired one used previously.
+
```
Comment (PastaPastaPasta) [inline doc/release-notes.md:45]:
```suggestion

# Removal of Non-Deterministic InstantSend

Legacy InstantSend was deprecated with the activation of Deterministic InstantSend but we kept the underlying logic for some time to ensure a smooth transition. v20.1 finally removes all Legacy InstantSend logic and will no longer support sending and receiving islock p2p message and inv p2p message with id equal 30. Such messages will be seen as unknown by v20.1 nodes. The protocol version was incremented to 70231 to indicate a change in network protocol. Please note that the -llmqinstantsend and -llmqtestinstantsend command line options have also been removed as part of this change.

```

**Review by knst — APPROVED** [review]
LGTM

**Review by UdjinM6 — APPROVED** [review]
utACK

File: doc/release-notes.md [hunk 16]
```diff
@@ -43,6 +43,10 @@ private keys, ensuring flexibility in managing their funds.
 
 Windows releases are signed by a new certificate that replaces the expired one used previously.
 
+# Removal of Non-Deterministic InstantSend
+
+Legacy InstantSend was deprecated with the activation of Deterministic InstantSend but we kept the underlying logic for some time to ensure a smooth transition. v20.1 finally removes all Legacy InstantSend logic and will no longer support sending and receiving islock p2p message and inv p2p message with id equal 30. Such messages will be seen as unknown by v20.1 nodes. The protocol version was incremented to 70231 to indicate a change in network protocol. Please note that the -llmqinstantsend and -llmqtestinstantsend command line options have also been removed as part of this change.
```
Comment (thephez) [inline doc/release-notes.md:48]:
```suggestion
## Removal of Non-Deterministic InstantSend

Legacy InstantSend was deprecated with the activation of Deterministic InstantSend, but we kept the underlying logic for some time to ensure a smooth transition. v20.1 finally removes all Legacy InstantSend logic and will no longer support sending and receiving islock p2p messages and inv p2p messages with id 30. Such messages will be seen as unknown by v20.1 nodes. The protocol version was incremented to 70231 to indicate a change in network protocol. Please note that the -llmqinstantsend and -llmqtestinstantsend command line options have also been removed as part of this change.
```

**Review by thephez — COMMENTED** [review]
The deterministic IS section is at the wrong heading level. Also, fixed a couple typos in that section.

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Review by thephez — APPROVED** [review]
utACK
